### PR TITLE
Made `label` optional in the parse strategy

### PR DESCRIPTION
**Pull request by jesper-friis**
# Description:
Made label optional in the parse strategy. Added unused metadata field.

## Type of change:
- [ ] Bug fix.
- [ ] New feature.
- [x] Documentation update.

## Checklist for the reviewer:

This checklist should be used as a help for the reviewer.

- [ ] Is the change limited to one issue?
- [ ] Does this PR close the issue?
- [ ] Is the code easy to read and understand, including clearly named variables?
- [ ] Do all new feature have an accompanying new test?
- [ ] Has the documentation been updated as necessary?

### CODE REVIEW

File: oteapi_dlite/strategies/parse.py [hunk 1]
```diff
@@ -52,9 +52,22 @@ class DLiteParseConfig(AttrDict):
...
+            ),
+        ),
+    ]
+    datamodel: Annotated[
+        Optional[str],
+        Field(
+            description=(
+                "DLite datamodel documenting the structure of the data set. "
+                "Often unused, since the datamodel is implicitly defined in "
+                "the strategy, but for a documentation point of view this is "
```
Comment (francescalb) [inline oteapi_dlite/strategies/parse.py:68]:
```suggestion
                "the DLite driver (DLite plugin), but for a documentation point of view this is "
```

File: oteapi_dlite/strategies/parse.py [hunk 2]
```diff
@@ -52,9 +52,22 @@ class DLiteParseConfig(AttrDict):
...
+        ),
+    ]
+    datamodel: Annotated[
+        Optional[str],
+        Field(
+            description=(
+                "DLite datamodel documenting the structure of the data set. "
+                "Often unused, since the datamodel is implicitly defined in "
+                "the DLite driver (DLite plugin), but for a documentation point of view this is "
+                "a very important field."
```
Comment (francescalb) [inline oteapi_dlite/strategies/parse.py:69]:
```suggestion
                "the DLite driver (DLite plugin), but for a documentation "
                "point of view this is a very important field."
```

**Review by francescalb — APPROVED** [review]

**Review by francescalb — CHANGES_REQUESTED** [review]
I orignially approved, but it does not work. I have added a fix.

However, there was not test for this core strategy, and this is something to be remedied immediately. I have started on a test, but please complete it.

File: tests/entities/Molecule.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+da31ff66-98af-590c-855d-386c63617755:
```
Comment (jesper-friis) [inline tests/entities/Molecule.yaml:1]: Can be removed - the generic yaml serialisation don't have this double-nesting.

**Review by francescalb — APPROVED** [review]
good

**Pull request merged**